scipy/interpolate/interpolate.py

```diff
@@ -2410,66 +2413,100 @@ class RegularGridInterpolator(object):
     # this class is based on code originally programmed by Johannes Buchner,
     # see https://github.com/JohannesBuchner/regulargrid
 
-    def __init__(self, points, values, method="linear", bounds_error=True,
-                 fill_value=np.nan):
-        if method not in ["linear", "nearest"]:
-            raise ValueError("Method '%s' is not defined" % method)
-        self.method = method
-        self.bounds_error = bounds_error
+    def __init__(self, points, values=None, xi=None, method="linear",
```

Comment:
I think placing a new keyword argument between current keyword arguments causes backward compatibility issues. If the overall idea seems useful to domain experts it may be necessary to move the new kwarg to the end.

Reply:
Yes it would cause backward compatibility issues because positional call of keyword arguments are allowed.
Now I use *args and **kwargs to deal with this limitation and ensure backward compatibility.
I also added extra tests and a doc test to show the utility of the new feature.